libraries/botbuilder-core/botbuilder/core/memory_storage.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from typing import Dict, List
+from copy import deepcopy
+from .storage import Storage, StoreItem
+
+
+class MemoryStorage(Storage):
+    def __init__(self, dictionary=None):
+        super(MemoryStorage, self).__init__()
+        self.memory = dictionary or {}
+        self._e_tag = 0
+
+    async def delete(self, keys: List[str]):
+        try:
+            for key in keys:
+                if key in self.memory:
+                    del self.memory[key]
+        except Exception as e:
+            raise e
+
+    async def read(self, keys: List[str]):
+        data = {}
+        try:
+            for key in keys:
+                if key in self.memory:
+                    data[key] = self.memory[key]
+        except Exception as e:
```

Comment:
Changing this line to `except TypeError as e:` to capture when keys is not iterable to avoid catching on the `Exception` class.